### ISSUE TITLE
Re-implement light theme

### ISSUE THREAD

**Issue by petitbambi** [PlusInsta/discord-plus#16]
I know this isn't a feature everyone uses, but I love having light / pastel backgrounds and having a grey / black text field on top of them just looks really ugly.

The light theme would work way better for them and allow the font to be colored black thus readable on a light background, but as is switching to the light theme only turns a couple of elements white and leaves most of them black.
 
Alternatively I've seen some themes with code that lets you recolor the grey 'text, server, member list, etc' fields to whichever color you want & change their transparency, but then we'd also need the option to change the font color so white text on top of a white text box doesn't become unreadable. (I've tried inserting the custom code that allows to change contrasts / opacity but if set to low the text just becomes unreadable due to matching the bg color)

I hope this isn't too big or difficult a task / request, I've tried a lot of themes but this is genuinely one of the best and visibly made with a lot of care - at this point it's the only I'm considering using but I can't seem to make it look good with my usual aesthetic / backgrounds and the dark overlays.

Thank you very much for your hard work so far & if you decide to implement this !

**Comment by PlusInsta** [comment]
I've thought about this for a while, but the biggest thing keeping me from simply reimplementing it is that the code becomes a mess. I've thought about two ways to go about it:
• Make a separate light mode branch
• Once BetterDiscord v2 rolls out, make the light theme code its own `.scss` file in addition to the dark theme file

The second option sounds more manageable, but that would require learning how the BDv2 theme engine works, and practically rewrite the theme to work with it.

**Comment by petitbambi** [comment]
Since the current version is basically a dark theme exclusively, light theme basically untouched version (due to its broken state) - would it be a lot of editing / rewriting to make a version that affects only the light mode - so basically the same code that it is now only with all the code indicating that it changes the dark theme switched to it affecting only the light theme ?

Sorry I only know basic code so I'm unsure how much work it would take & if any of it is reusable 

After a lot of tinkering I realized that the 'Increase contrast' code in FAQ actually allows for recoloring the fields as well, but again the text stays a light grey - even being able to change it to just pure white or pure black would help readability on lighter backgrounds & transparency settings but I haven't been able to find anyone who has implemented something like this into their theme other than simply using light theme to change the text to black - and no basic code that I know is capable of affecting it 

either way I appreciate your efforts and explanation !

**Comment by PlusInsta** [comment]
I won't be re-using any light theme code. Most of it is ancient, buggy, hacky, or otherwise difficult to maintain. That's the reason I removed so much of it; it's simply not salvageable in the long run. 7e16b71e4a481fb7016e56d525301f8c4e191c5d alone took me a while to commit because of how inconsistent the light mode code was when compared to the dark theme code. **Tl;dr** my experience with CSS was limited, and as such, the code sucks.

One of the things that made my theme stand out from the original light theme is that the left side panel was also light. To pull that off, I had to use many hacks, such as inverting icon colors. It's quick and dirty, but it worked. And by "quick and dirty," I mean it's the closest thing I could do without losing my damn mind over how annoying Discord's light mode CSS is to work with.

To put it simply, if I'm going to re-implement light mode, **I'll rewrite everything from scratch** so the code can be easily maintained. But to do that, **I'll have to wait until BDv2 releases** so the code is **easy to maintain within its own `.scss` file**.

> That's totally not just me being lazy or anything

**Comment by petitbambi** [comment]
Oh I see, I totally understand now - sorry for bothering but thank you for taking it into consideration !

**Comment by PlusInsta** [comment]
Nah, it's fine! I've been looking for an excuse to re-implement the light theme for ages. So far, the only thing keeping me from doing it is "nobody wants this."

I'll just leave the issue open for the day BetterDiscord v2 rolls out.

**Comment by PlusInsta** [comment]
A blessing from the heavens came, and Discord updated their light theme code _**at last**_. It's now at least 10000% easier to deal with thanks to the built-in variables.

# Light theme support is happening.

![Screenshot](https://user-images.githubusercontent.com/38794586/64682002-facaae00-d480-11e9-9a80-b82fcb765264.png)

It's still a work-in-progress, but thanks to how streamlined the code is now, it's gonna be much, MUCH easier to maintain.

**Comment by PlusInsta** [comment]
Light theme has been implemented for a while now, so I'll go ahead and close this.
